### DISCUSSION TITLE
Once per week vs. every 7 days

### DISCUSSION THREAD

**Discussion by MelissaBo13** [iSoron/uhabits#2023] [Help & FAQ]
Hi! Whenever I set a habit to once per week, it changes it to "every 7 days" when I save it. This isn't the same thing- I want it to just show the empty checkmarks for the rest of that calendar week, but instead it's showing it for a full seven days, no matter where it is in the week. Is this a glitch?

**Comment by KyleSCraig** [comment]
Check out issue #1999
I'm interested in this for the month case where you can't do every 30 or 31 days but I do use both. And indeed I can see the code doesn't look to be easy to modify to have both situations work.

**Reply by MelissaBo13** [reply]
Wait I just realized it's also doing this for my 3x per week habits. It should be starting new today, but it's showing an empty checkmark. Is everything on this app on a rolling week?

**Reply by KyleSCraig** [reply]
Yes although per month (30 or 31 days) does seem to have special behavior. Normally if you do it on a day of the month, it becomes due again on that same day of the month. If you do it on the last day of the month it has a special behavior where it becomes due on the last day the next month.
